src/morphdom.js

```diff
@@ -300,6 +300,7 @@ export default function morphdomFactory(morphAttrs) {
                                             }
 
                                             curFromNodeChild = matchingFromEl;
+                                            curFromNodeKey = getNodeKey(curFromNodeChild);
```

Comment:
When `curFromNodeChild` is re-assigned here, `curFromNodeKey` was out of sync.

This caused the wrong key to be scheduled for deferred removal on line 350 if the `matchingFromEl` can't be morphed to `curToNodeChild`, such as when the DOM ids match but the element names are different.